x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/action/StartTrainedModelDeploymentAction.java

```diff
@@ -623,6 +623,9 @@ public String getDeploymentId() {
          * @return the estimated memory (in bytes) required for the model deployment to run
          */
         public long estimateMemoryUsageBytes() {
+            if (numberOfAllocations == 0) {
```

Comment:
This method is on `TaskParams` - `StartTrainedModelDeploymentAction.TaskParams. estimateMemoryUsageBytes()`

There is another public method `StartTrainedModelDeploymentAction.estimateMemoryUsageBytes()` on line 792 that needs this check. 

If StartTrainedModelDeploymentAction.estimateMemoryUsageBytes() can return - then line 635 ` + (cacheSize.getBytes() - modelBytes);` needs a Max.(0,...) to ensure the return is non-negative

Reply:
Fixed `StartTrainedModelDeploymentAction.estimateMemoryUsageBytes`

I don't think the `Max` is necessary. `StartTrainedModelDeploymentAction.estimateMemoryUsageBytes` returns 0 only if the number of allocations is 0, in which case the `TaskParams.estimateMemoryUsageBytes` already returns 0.